src/Client.php

```diff
@@ -156,20 +164,24 @@ public function flush(?int $timeout = null): PromiseInterface
     /**
      * Assembles an event and prepares it to be sent of to Sentry.
      *
-     * @param array      $payload the payload that will be converted to an Event
-     * @param Scope|null $scope   optional scope which enriches the Event
+     * @param array      $payload The payload that will be converted to an Event
+     * @param Scope|null $scope   Optional scope which enriches the Event
      *
      * @return Event|null returns ready to send Event, however depending on options it can be discarded
      */
     private function prepareEvent(array $payload, ?Scope $scope = null): ?Event
     {
-        $sampleRate = $this->getOptions()->getSampleRate();
+        $sampleRate = $this->options->getSampleRate();
 
         if ($sampleRate < 1 && mt_rand(1, 100) / 100.0 > $sampleRate) {
+            if (null !== $this->logger) {
+                $this->logger->info('The event will be discarded because it has been sampled.');
```

Comment:
Maybe we should add the event itself (or at least some data about it) in the logger's context?

Reply:
Indeed it makes sense. I've added the event object to the context data